Bottom: a4b09c3cdc7c25dd763ce76138432c0aba19fc68
Top:    c64a76f16b2f730e7c7514ef9ef676f4e60f07e0
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-07-10 17:35:28 -0700

Refresh of build-preload

---

diff --git a/Makefile.am b/Makefile.am
index d74bbac..e01b4cc 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,14 +8,15 @@ ACLOCAL_AMFLAGS = -I config
 AM_CFLAGS = -g -Wall -D_GNU_SOURCE -DSYSCONFDIR=\"$(sysconfdir)\" -DRDMADIR=\"@rdmadir@\"
 
 src_librdmacm_la_CFLAGS = $(AM_CFLAGS)
-src_librspreload_la_CFLAGS = -Wall -ldl -shared -fPIC -D_GNU_SOURCE
+src_librspreload_la_CFLAGS = $(AM_CFLAGS)
 
 if HAVE_LD_VERSION_SCRIPT
     librdmacm_version_script = -Wl,--version-script=$(srcdir)/src/librdmacm.map
+    librspreload_version_script =
 else
     librdmacm_version_script =
+    librspreload_version_script =
 endif
-librspreload_version_script =
 
 src_librdmacm_la_SOURCES = src/cma.c src/addrinfo.c src/acm.c \
 		src/rsocket.c src/indexer.c
diff --git a/src/preload.c b/src/preload.c
index 5ed77d7..8758d5c 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -31,14 +31,6 @@
  *
  */
 
-/*
- * Build:
- * gcc -lrdmacm -Wall -shared -fPIC -o preload.so -D_GNU_SOURCE preload.c indexer.c
- *
- * Use:
- * LD_PRELOAD=preload.so binary --plus_args
- */
-
 #if HAVE_CONFIG_H
 #  include <config.h>
 #endif /* HAVE_CONFIG_H */
@@ -54,7 +46,6 @@
 #include <string.h>
 #include <netinet/in.h>
 #include <netinet/tcp.h>
-#include <stdio.h>
 
 #include <rdma/rdma_cma.h>
 #include <rdma/rdma_verbs.h>
@@ -225,7 +216,6 @@ static void init_preload(void)
 	/* Quick check without lock */
 	if (init)
 		return;
-printf("%s\n", __func__);
 
 	pthread_mutex_lock(&mut);
 	if (init)
@@ -330,7 +320,6 @@ int socket(int domain, int type, int protocol)
 	static __thread int recursive;
 	int index, ret;
 
-printf("%s\n", __func__);
 	if (recursive)
 		goto real;
